modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -0,0 +1,676 @@
+                    cv::gimpl::createMat(cv::util::get<cv::GMatDesc>(out_metas[id]), newMat);
+                    out_data[id] = cv::GRunArg(std::move(newMat));
+                    isl_outputs[id] = { r, cv::GRunArgP(&cv::util::get<MatType>(out_data[id])) };
+                }
+                break;
+            case cv::GShape::GSCALAR:
+                {
+                    SclType newScl;
+                    out_data[id] = cv::GRunArg(std::move(newScl));
+                    isl_outputs[id] = { r, cv::GRunArgP(&cv::util::get<SclType>(out_data[id])) };
+                }
+                break;
+            case cv::GShape::GARRAY:
+                {
+                    cv::detail::VectorRef newVec;
+                    cv::util::get<cv::detail::ConstructVec>(r.ctor)(newVec);
+                    out_data[id]= cv::GRunArg(std::move(newVec));
+                    // VectorRef is implicitly shared so no pointer is taken here
+                    const auto &rr = cv::util::get<cv::detail::VectorRef>(out_data[id]); // FIXME: that variant MOVE problem again
+                    isl_outputs[id] = { r, cv::GRunArgP(rr) };
+                }
+                break;
+            default:
+                cv::util::throw_error(std::logic_error("Unsupported GShape"));
+                break;
+            }
+        }
+        // Now ask Island to execute on this data
+        island->run(std::move(isl_inputs), std::move(isl_outputs));
+
+        // Once executed, dispatch our results down to the pipeline.
+        for (auto &&it : ade::util::zip(ade::util::toRange(out_queues),
+                                        ade::util::toRange(out_data)))
+        {
+            for (auto &&q : std::get<0>(it))
+            {
+                // FIXME: FATAL VARIANT ISSUE!!
+                const auto tmp = std::get<1>(it);
+                q->push(Cmd{tmp});
+            }
+        }
+    }
+}
+
+void collectorThread(std::vector<Q*> in_queues,
+                     Q&              out_queue)
+{
+    while (true)
+    {
+        cv::GRunArgs this_result(in_queues.size());
+        std::size_t stops = 0u;
+        for (auto &&it : ade::util::indexed(in_queues))
+        {
+            Cmd cmd;
+            ade::util::value(it)->pop(cmd);
+            if (cmd.index() == cmd.index_of<Stop>()) {
+                stops++;
+            } else {
+                // FIXME: MOVE_PROBLEM
+                const cv::GRunArg &in_arg = cv::util::get<cv::GRunArg>(cmd);
+                this_result[ade::util::index(it)] = in_arg;
+                // FIXME: Check for other message types.
+            }
+        }
+        if (stops > 0)
+        {
+            GAPI_Assert(stops == in_queues.size());
+            out_queue.push(Cmd{Stop{}});
+            return;
+        }
+        out_queue.push(Cmd{this_result});
+    }
+}
+} // anonymous namespace
+
+cv::gimpl::GStreamingExecutor::GStreamingExecutor(std::unique_ptr<ade::Graph> &&g_model)
```

Comment:
Why g_model?

Reply:
I believe it was a copy-paste from a general `GExecutor`, but why not?

Reply:
In fact we have two ADE Graphs: the low-level one (viewed as GModel) and a higher-level one (island-based, GIslandModel). This one stands to what we usually access via GModel, so here is the name